blog/_posts/2017-11-27-view-based-collections.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy
+title: Let Them Be Lazy!
+---
+
+One of the main changes in the new design of the collections is that operations have
+“view based” default implementations rather than “builder based”. This blog article
+explains what this means as well as what it means for a collection to be lazy or
+strict, how the view based design allows us to support both types of collections,
+and finally why this is an improvement over the current design of the collections.
+
+More background information about the collections redesign can be found in
+[this blog post](http://www.scala-lang.org/blog/2017/02/28/collections-rework.html). The
+source code is available in
+[this GitHub repository](https://github.com/scala/collection-strawman).
+
+## Background: Lazy vs Strict Collections
+
+We can classify collections in two categories: those that evaluate their elements when
+they are created, and those that evaluate their elements when they are needed only.
+
+An example of the first category is `List[A]`: when we create a list we compute all
+its elements, when we transform a list (e.g. using `map`) we compute all the elements
+of the resulting list. We say that list is a **strict** collection type.
+
+An example of the second category is `Stream[A]`: we can create and transform streams
```

Comment:
Mention that `Stream` will be replaced by `LazyList`?

Reply:
I don’t think this will help following the article.